tests/unit/src/main/java/org/mule/tck/core/internal/serialization/CustomByteArrayInputStream.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.core.internal.serialization;
+
+import java.io.IOException;
```

Comment:
This is just a comment (not sure about this): this is not a complete decorator. It can be used for testing purposes only in the case that the this specific read method is invoked (this will failed for not overridden methods). If it will not be generally used, I would declare it as an inner class in the test.